### PR TITLE
feat: add leap exercise

### PR DESCRIPTION
**Pull request by vaeng**

### CODE REVIEW

File: exercises/practice/leap/leap.sql
Comment (IsaacG) [inline exercises/practice/leap/leap.sql:1]: Should there be some sort of something here? Maybe provide the schema?

**Review by IsaacG — APPROVED** [review]

File: exercises/practice/leap/leap.sql
Comment (vaeng) [reply exercises/practice/leap/leap.sql:1]: You're right, I pushed a new commit that will show the csv and the create_fixture.sql file in the online editor as read only.

File: exercises/practice/leap/leap_test.sql [hunk 1]
```diff
@@ -0,0 +1,59 @@
+-- Create database:
+.read ./create_fixture.sql
+
+-- Read user student solution and save any output as markdown in user_output.md:
+.mode markdown
+.output user_output.md
+.read ./success.sql
```
Comment (IsaacG) [inline exercises/practice/leap/leap_test.sql:7]: Is this supposed to be `leap.sql` ?

Comment (vaeng) [reply exercises/practice/leap/leap_test.sql:7]: Yes. Bad copy and paste from the test runner. Thanks for the thorough check. We need to implement ci to run tests, so we know they are fibe

File: exercises/practice/leap/leap_test.sql [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+.output
+
+-- Create a clean testing environment:
+DROP TABLE IF EXISTS main.tests;
+CREATE TABLE IF NOT EXISTS main.tests (
```
Comment (IsaacG) [inline exercises/practice/leap/leap_test.sql:12]: The `tests` table is sometimes referred to with a namespace and other times without. Should that be consistent?

File: exercises/practice/leap/leap_test.sql [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    task_id INTEGER DEFAULT NULL,
+    -- Here are columns for the actual tests
+    year INT NOT NULL,
+    result BOOL NOT NULL
+);
+
+INSERT INTO tests (name, uuid, year, result)
+    VALUES
+        -- Every test case from the .meta/tests.toml file gets its own row:
+        ("year not divisible by 4 in common year", "6466b30d-519c-438e-935d-388224ab5223", 2015, 0),
```
Comment (IsaacG) [inline exercises/practice/leap/leap_test.sql:30]: My version of sqlite3 doesn't like `"` and requires `'`. I don't know how uniform that need is.

File: exercises/practice/leap/leap_test.sql (hunk 1 above)
Comment (IsaacG) [reply exercises/practice/leap/leap_test.sql:7]: Less a thorough review, more me trying to copy/paste and reuse for raindrops :smile:

File: exercises/practice/leap/leap_test.sql (hunk 3 above)
Comment (vaeng) [reply exercises/practice/leap/leap_test.sql:30]: When we set up the cli tests, we should try to check this behavior.
I have tested the double quotes locally with version 3.37.2 on ubuntu 22.04.
The test runner is fine with it on alpine 3.18 with sqlite3 version 3.41.2.

Comment (vaeng) [reply exercises/practice/leap/leap_test.sql:30]: But we should use single quotes:

> Use single-quotes, not double-quotes, around string literals in SQL. This is what the SQL standard requires. Your WHERE clause expression should read: column1='column1'

> SQL uses double-quotes around identifiers (column or table names) that contains special characters or which are keywords. So double-quotes are a way of escaping identifier names. Hence, when you say column1="column1" that is equivalent to column1=column1 which is obviously always true.

From https://www.sqlite.org/faq.html Item 24.

File: exercises/practice/hello-world/hello-world_test.sql [hunk 4]
```diff
@@ -25,18 +25,18 @@ CREATE TABLE IF NOT EXISTS main.tests (
...
 -- Comparison of user input and the tests updates the status for each test:
 UPDATE tests
-SET status = "pass"
+SET status = 'pass'
 FROM (SELECT greeting FROM hello_world) AS h
 WHERE h.greeting = tests.result;
 -- Upadte message for failed tests to give helpful information:
 UPDATE tests
-SET message = "Greeting is: '" || h.greeting || "', but should be: '" || tests.result ||"'"
+SET message = 'Greeting is: '' || h.greeting || '', but should be: '' || tests.result ||'''
```
Comment (IsaacG) [inline exercises/practice/hello-world/hello-world_test.sql:37]: The nested quote looks broken.

Comment (vaeng) [reply exercises/practice/hello-world/hello-world_test.sql:37]: Apparently ' inside a single quoted string must be escaped by another ': https://stackoverflow.com/a/70414773/4919081

**Pull request merged**